sdk/webpubsub/Azure.Messaging.WebPubSub/tests/Samples/WebPubSubSamples.HelloWorld.cs

```diff
@@ -91,7 +90,7 @@ public void AddUserToGroup()
             client.AddUserToGroup("some_group", "some_user");
 
             // Avoid sending messages to users who do not exist.
-            if (client.UserExists("some_user"))
+            if (client.UserExists("some_user", CancellationToken.None).Value)
```

Comment:
Is the `CancellationToken` required to bind?  Do we want convenience wrappers for APIs without any parameters to disambiguate?

Reply:
It is necessary, because today we generate RequestOptions as always defaultable.  We had discussed making the parameter required in cases where the operation didn't take a body (if it took a body then the protocol method would be a RequestContent parameter and then the compiler could work out that you didn't want that overload) to make it so that if we added a grow-up version the compiler would prefer it.

@KrzysztofCwalina pointed out that it will make these methods hard to call and that for many of them they may never "grow up" and so we'd have a bit of a wart.

There is a tension here, and I think we have to make a call.

The one thing we did discuss is for cases like this (where we decide we want to grow up some method before shipping a GA version) we could generate the protocol versions with a required `RequestOptions` parameter, so that it didn't lead to ambiguities.

For this specific issue, I'm not sure if we should just teach the code generator about these sort of `XyzExists` type methods and have a way to denote in the swagger (this operation is a logical existence check, return Response<bool> where the bool is checking for some known family of status values.

Reply:
I'd prefer the "required `RequestOptions` when there's no body" by default and only allow it to be relaxed in cases where we've explicitly thought about the scenario enough to make it optional.

I also think an `x-ms-returns-successful` extension or similar might be worth creating, but I haven't seen it in too many services.  This one happens to use it more than most.